monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,133 @@
+from pathlib import Path
+from typing import  Any
+from utils import OperatorLocation, RIDFlightDetails, TestFlightDetails, TestFlight
+from urllib.parse import urlparse
+from monitoring.monitorlib.rid import SCOPE_READ, SCOPE_WRITE
+import time
+
+class TestBuilder():
+    ''' A class to setup the test data and create the objects ready to be submitted to the test harness '''
+
+    def __init__(self, test_config: str, country_code='che') -> None:
+        
+        self.test_config_valid(test_config)
+        self.test_config = json.loads(test_config)
+        self.tracks_directory = Path('test_definitions', country_code, 'aircraft_states')
+        self.verify_tracks_directory(self.tracks_directory)
+        self.flight_tracks = self.load_flight_tracks(self.tracks_directory)
+            
+    def load_flight_tracks(self, tracks_directory) -> None:
+        track_files = os.listdir(tracks_directory) 
+        return track_files
+
+    def verify_tracks_directory(self, tracks_directory) -> None:
+
+        ''' This method checks if there are tracks in the tracks directory '''        
+        
+        files = [f for f in os.listdir(tracks_directory) if os.path.isfile(os.path.join(tracks_directory, f))]
+        if files:
+            pass
+        else:
+            raise ValueError("The there are no tracks in the tracks directory, create tracks first using the flight_data_generator module. ")
+
+    def test_config_valid(self, test_config: str) -> None:
+        ''' This method checks if the test definition is a valid JSON ''' #TODO : Have a comprehensive way to check JSON definition
+        if json.loads(test_config):
+            pass
+        else:
+            raise ValueError("A valid JSON object must be submitted ")
+
+    def make_json_compatible(self, struct: Any) -> Any:
+        if isinstance(struct, tuple) and hasattr(struct, '_asdict'):
+            return {k: self.make_json_compatible(v) for k, v in struct._asdict().items()}
+        elif isinstance(struct, dict):
+            return {k: self.make_json_compatible(v) for k, v in struct.items()}
+        elif isinstance(struct, str):
+            return struct
+        try:
+            return [self.make_json_compatible(v) for v in struct]
+        except TypeError:
+            return struct
+
+    def build_test_payload(self): 
+        ''' This is the main method to process the test configuration and build RID payload object, maxium of one flight is allocated to each USS. '''
+        
+        usses = self.test_config['usses']
+
+        all_test_payloads = []
+        
+        for uss_index, uss in enumerate(usses):
+            requested_flights = []
+            flight_track_path = Path(self.tracks_directory, self.flight_tracks[uss['allocated_flight_track_number']])
+            with open(flight_track_path) as generated_rid_state:
+                rid_state_data = json.load(generated_rid_state)
+            
+            effective_after = rid_state_data['reference_time']
```

Comment:
While we want aircraft state replayer to do as little dynamic data modification or generation, one thing it must do to function reasonably is to adjust all the timestamps.  If no adjustment is performed, we will be injecting data into a USS that was stale long ago (since the timestamps stored on disk were generated by the simulator long ago).

Reply:
OK, agreed, I have now updated the timestamps.